### PR TITLE
fix(esl-mixin-element): major fix for nested hierarchy mixin disconnection

### PR DESCRIPTION
**Pull request by ala-n**
Closes: #2505

### CODE REVIEW

File: src/modules/esl-mixin-element/ui/esl-mixin-registry.ts [hunk 1]
```diff
@@ -170,9 +170,15 @@ export class ESLMixinRegistry {
...
-  private static destroyAll(el: HTMLElement): void {
+  /**
+   * Destroys all mixins on the element
+   * @param el - host element to destroy all mixins
+   * @param deep - if true, will destroy all mixins on the subtree
+   */
+  private static destroyAll(el: HTMLElement, deep = true): void {
     const store = (el as any)[STORE] as Record<string, ESLMixinElement> | undefined;
     store && Object.keys(store).forEach((name) => ESLMixinRegistry.destroy(el, name));
+    if (!deep || !el.children || !el.children.length) return;
```
Comment (abarmina) [inline src/modules/esl-mixin-element/ui/esl-mixin-registry.ts]: can we simplify it?
```suggestion
    if (!deep || !(el.children?.length)) return;
```

**Review by abarmina — APPROVED** [review]

File: src/modules/esl-mixin-element/ui/esl-mixin-registry.ts [hunk 2]
```diff
@@ -170,9 +170,11 @@ export class ESLMixinRegistry {
     delete store[mixin];
   }
 
-  /** Destroys all mixins on the element */
+  /** Destroys all mixins on the element and its subtree */
   private static destroyAll(el: HTMLElement): void {
     const store = (el as any)[STORE] as Record<string, ESLMixinElement> | undefined;
     store && Object.keys(store).forEach((name) => ESLMixinRegistry.destroy(el, name));
+    if (!el.children || !el.children.length) return;
```
Comment (abarmina) [inline src/modules/esl-mixin-element/ui/esl-mixin-registry.ts:177]:
```suggestion
    if (!(el.children?.length) return;
```

**Review by dshovchko — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Pull request merged**